src/Platforms/MariaDb1043Platform.php

```diff
@@ -76,30 +76,33 @@ public function getListTableColumnsSQL($table, $database = null): string
      *
      * @return array{string, string}
      */
-    public function getColumnTypeSQLSnippets(string $tableAlias = 'c'): array
+    public function getColumnTypeSQLSnippets(string $tableAlias = 'c', ?string $databaseName = null): array
     {
         if ($this->getJsonTypeDeclarationSQL([]) !== 'JSON') {
-            return parent::getColumnTypeSQLSnippets($tableAlias);
+            return parent::getColumnTypeSQLSnippets($tableAlias, $databaseName);
         }
 
+        $databaseName = $this->getDatabaseNameSQL($databaseName);
+
         $columnTypeSQL = <<<SQL
             IF(
-                x.CHECK_CLAUSE IS NOT NULL AND $tableAlias.COLUMN_TYPE = 'longtext',
+                $tableAlias.COLUMN_TYPE = 'longtext'
+                AND EXISTS(
+                    SELECT * from information_schema.CHECK_CONSTRAINTS 
+                    WHERE CONSTRAINT_SCHEMA = $databaseName
```

Comment:
I tried using `WHERE CONSTRAINT_SCHEMA = $tableAlias.TABLE_SCHEMA` here to simplify the PHP code, but sadly this seems to prevent the optimization in MariaDB.